### PR TITLE
Issue 253: Added question functionality to instructions

### PR DESCRIPTION
**Pull request by JRustyHaner**
#253 
using TDFs, a yes/no question can be displayed during /instructions interstitial. The results will be stored in the Session variable instructionQuestionResults, which can be passed into the p value calculations.

### CODE REVIEW

File: mofacts/client/views/experiment/instructions.js [hunk 1]
```diff
@@ -295,11 +295,23 @@ Template.instructions.helpers({
...
   },
-
-  instructions: function() {
-    return _.chain(Session.get('currentTdfUnit')).prop('unitinstructions').trim().value();
+  instructionText: function(){
+    text = Session.get('currentTdfFile').tdfs.tutor.unit[0].unitinstructions;
+    return text;
+  },
+  instructionQuestion: function(){
+    question = Session.get('currentTdfFile').tdfs.tutor.unit[0].unitinstructionsquestion;
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/instructions.js:303]: Doubt this should be hardcoded

**Review by andrewtackett — CHANGES_REQUESTED** [review]
Did you update the documentation?

Also, I don't see any integration of the result back into the p parameters

**Comment by imrryr** [comment]
I agree it shouldn't be hard coded

**Comment by imrryr** [comment]
I'll try to think of a better alternative that fits future goals.

**Comment by JRustyHaner** [comment]
I removed the hard coding, it was throwing me for a loop because each unit has the possibility of throwing instructions, although card.js misses the first <unit> tag. Is that intentional? It could just check for if the unit[0] has unitinstructions defined. As a point to clarify, how should the question affect the p value, or should it be just stored?

**Comment by imrryr** [comment]
So that definitely fulfills what we need however this ability to transfer
information from one unit to another is very powerful and we should
consider a future generalization of the ability. It sounds like it would
make sense to make that a different issue though unless what you have
created generalizes very easy...

**Comment by JRustyHaner** [comment]
ideally, if I were to generalize it further, I would allow for an array of questions ("On a scale of 1 to 5, how difficult is this material to understand?")to be asked with a variety of response types, ie, yes/no, multi choice, etc. Since these questions do not count against the scoring, it could be used to gather other data that the model could use as predictions. Sort of like a pre-exam questionaire.

**Comment by imrryr** [comment]
Yes that's exactly the way it will be used after the grant in future work.
Clearly there's no big hurry on that then but it might be nice to discuss
tomorrow

**Comment by JRustyHaner** [comment]
@andrewtackett

File: mofacts/client/views/experiment/instructions.js [hunk 2]
```diff
@@ -340,4 +340,14 @@ Template.instructions.events({
     event.preventDefault();
     instructContinue();
   },
+  'click #instructionQuestionAffrimative': function() {
+    Session.set('instructionQuestionResults',true);
+    $('#instructionQuestion').hide();
+
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/instructions.js:346]: extra whitespace

File: mofacts/client/views/experiment/instructions.js [hunk 3]
```diff
@@ -340,4 +340,14 @@ Template.instructions.events({
...
+  },
+  'click #instructionQuestionNegative': function() {
+    Session.set('instructionQuestionResults',false);
+    $('#instructionQuestion').hide();
+
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/instructions.js:351]: extra whitespace

File: mofacts/client/views/experiment/unitEngine.js [hunk 4]
```diff
@@ -903,6 +912,7 @@ function modelUnitEngine() {
           priorStudy: card.priorStudy,
           totalPracticeDuration: card.totalPracticeDuration,
           outcomeStack: card.outcomeStack.join(','),
+          instructionQuestionResult: session.get('instructionQuestionResult'),
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/unitEngine.js:915]: Does this work with lowercase session?

**Pull request merged**